src/lv_themes/lv_theme_material.c

```diff
@@ -195,7 +192,7 @@ static bool inited;
 
 static void basic_init(void)
 {
-    style_init_reset(&styles->scr);
+    lv_style_reset(&styles->scr);
```

Comment:
On the first call  `lv_style_reset` is called on an uninitialized variable that contains garbage.

Therefore [lv_mem_free](https://github.com/lvgl/lvgl/pull/2074/files#diff-9db09d73399ff687dc1e13ed3fd25c8d5f8a609bed893784193dbcf825b6b9f8R324) attempts to free a pointer whose value is garbage.

You can try it by setting [LV_MEM_ADD_JUNK](https://github.com/lvgl/lvgl/blob/87a8be362431a89a2fdd4c74defe454523656f9f/src/lv_misc/lv_mem.c#L24) to 1.

Reply:
> On the first call `lv_style_reset` is called on an uninitialized variable that contains garbage.
> 

No, the patch call _lv_memset_00 on the allocated memory:
https://github.com/lvgl/lvgl/pull/2074/files#diff-d342b8ca626c1740e2afd1b85728842b0aa8cf8ca03a445b1f56e109154f743cR70

> Therefore [lv_mem_free](https://github.com/lvgl/lvgl/pull/2074/files#diff-9db09d73399ff687dc1e13ed3fd25c8d5f8a609bed893784193dbcf825b6b9f8R324) attempts to free a pointer whose value is garbage.
> 

Since all fields in theme_styles_t get zerod, lv_mem_free will skip here directly:
https://github.com/lvgl/lvgl/blob/87a8be362431a89a2fdd4c74defe454523656f9f/src/lv_misc/lv_mem.c#L221

> You can try it by setting [LV_MEM_ADD_JUNK](https://github.com/lvgl/lvgl/blob/87a8be362431a89a2fdd4c74defe454523656f9f/src/lv_misc/lv_mem.c#L24) to 1.

Reply:
Ah, true. Thanks for the explanation.